src/vm/finalizerthread.cpp

```diff
@@ -891,6 +913,11 @@ void FinalizerThread::FinalizerThreadCreate()
     hEventShutDownToFinalizer = new CLREvent();
     hEventShutDownToFinalizer->CreateAutoEvent(FALSE);
 
+#ifdef FEATURE_UNIX_LOW_MEMORY_NOTIFICATION
+    _ASSERTE(s_pLowMemoryDetector == 0);
+    s_pLowMemoryDetector = new UnixLowMemoryDetector();
```

Comment:
I would prefer to not to introduce more points where we can fail due to not enough memory. We don't need to allocate here if we change it as follows. Could you please change the UnixLowMemoryDetector so that it would keep the memory limit in a static member, replace the constructor with a static `Init` method that would do what the constructor used to do and finally change the s_pLowMemoryDetector to be a static member of the FinalizerThread?